Tiles/RedChimneyTE.cs

```diff
@@ -27,26 +28,23 @@ public enum State : byte
 		
 		public override void Update()
 		{
-			if(CurrentState == State.Smoke && Main.rand.Next(9) == 0)
+			if(CurrentState == State.Smoke && Main.rand.NextBool(9))
 			{
-				var smokePos = new Vector2(Position.X * 16 + 16, Position.Y * 16 + 8);
-				var smokeVel = Vector2.Zero;
-				if (Main.windSpeedCurrent < 0f)
+				Vector2 smokePos = new (Position.X * 16 + 16, Position.Y * 16 + 8);
+				Vector2 smokeVel = Vector2.Zero;
+				int smokeType = Main.rand.Next(GoreID.ChimneySmoke1, GoreID.ChimneySmoke3);
+				byte rand = (byte)Main.rand.Next(4);
+				if (rand == 3)
 				{
-					smokeVel.X = -Main.windSpeedCurrent;
+					Gore.NewGore(new EntitySource_TileEntity(this), smokePos, smokeVel, smokeType, Main.rand.NextFloat() * 0.4f + 0.4f);
 				}
```

Comment:
The if state and the change in velocity are what became redundant, cause their behavior is harcoded